### DISCUSSION TITLE
No reverb or chorus effect

### DISCUSSION THREAD

**Discussion by jbigdata-git** [FluidSynth/fluidsynth#900] [Q&A]
Hi,

I can't ear effects as reverb or chorus.

**FluidSynth runtime version 2.2.1 + ALSA on Ubuntu**

Here is the C snippet

> settings = new_fluid_settings();
> int res = fluid_settings_setstr(settings, "audio.driver", "alsa");
> if (res != FLUID_OK)
> {
>   puts("adriver reg error");
>   return -1;
> }
> synth = new_fluid_synth(settings);
> res = fluid_synth_chorus_on(synth, -1, 1);
> res = fluid_synth_set_chorus_group_level(synth, -1, 1.2);
> res = fluid_synth_set_chorus_group_speed(synth, -1, 0.3);
> res = fluid_synth_set_chorus_group_depth(synth, -1, 8);
> res = fluid_synth_set_chorus_group_type(synth, -1, FLUID_CHORUS_MOD_SINE);
> res = fluid_synth_set_chorus_group_nr(synth, -1, 20);

All _res_ are FLUID_OK.
The midi file sounds well but effect set or not, the sound remains the same.
I also use **Qsynth 0.9.3**, the only button having an effect is _Gain_. Tweaking chorus and reverb buttons produces no effect.

I found that **soundfont** file used is the key.
I use classic _FluidR3_GM.sf2_ but didn't found an out of the box sf2 file that contains piano effects.
Keep on my way to use effects with _fluidsynth_, pointers or tutorials are welcome.

JB

**Comment by be1** [comment]
I have the same problem, trying to set reverberation effect:
all returns are FLUID_OK but no reverb is heard despite levels are at their max.

**Reply by be1** [reply]
The fluidsynth documentation says: «Note that the amount of signal sent to the reverb module depends on the "reverb send" generator defined in the SoundFont.» Does this mean we do not have such a soundfont defining the "reverb send" generator ?

**Comment by ppyne** [comment]
For my part the problem of the missing reverb and chorus in audio output was due to the two unset generator parameters :

- GEN_CHORUSSEND: Chorus send amount.
- GEN_REVERBSEND: Reverb send amount.

in the sf2 soundfont.
